src/commands/block/addon.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable max-depth */
+import { Flags } from "@oclif/core";
+import {
+  combinedDiggerJson,
+  createAddon,
+  createOrUpdateVpc,
+  diggerJson,
+} from "../../utils/helpers";
+import { trackEvent } from "../../utils/mixpanel";
+import { BaseCommand } from "../../base";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import * as inquirer from "inquirer-shortcuts";
+
+export default class Addon extends BaseCommand<typeof Addon> {
+  static description = "Addon to a block";
+
+  static flags = {
+    type: Flags.string({
+      char: "t",
+      description: "type of addon",
+      options: ["routing", "regions"],
+    }),
+    block: Flags.string({
+      char: "b",
+      description: "block to add the addon to",
+      required: true,
+    }),
+  };
+
+  public async run(): Promise<void> {
+    const { args, flags } = await this.parse(Addon);
+    trackEvent("block addon called", { flags, args });
+    if (!flags.type) {
+      this.log(
```

Comment:
Just a though I had while working on block-only init - why don't we move to a pattern that where the user has to choose from predefined options, we just prompt to choose if it's not passed as a flag. This would require less re-runs for the user to get it right first time they run the command. We could even consider printing out what command they ran (with the prompt filled in)

```
const { type } = flags.type
        ? { type: flags.type }
        : await inquirer.prompt([
            {
              type: "list",
              name: "type",
              message: "Select type of addon",
              choices: ["routing", "regions"],
            },
          ]);
```

Reply:
I was thinking about it, but it's gonna be messy in this case, where you need to provide multiple answers per option. Will revisit it again later